data/signingrecord/src/main/java/tech/pegasys/teku/data/signingrecord/ValidatorSigningRecord.java

```diff
@@ -119,9 +119,36 @@ public Bytes toBytes() {
               + genesisValidatorsRoot);
       return Optional.empty();
     }
-    if (isSafeSourceEpoch(sourceEpoch) && isSafeTargetEpoch(targetEpoch)) {
+
+    boolean sourceEpochIsSafe = isSafeSourceEpoch(sourceEpoch);
+    boolean targetEpochIsSafe = isSafeTargetEpoch(targetEpoch);
+    if (sourceEpochIsSafe && targetEpochIsSafe) {
       return Optional.of(
           new ValidatorSigningRecord(genesisValidatorsRoot, blockSlot, sourceEpoch, targetEpoch));
+    } else {
+      if (!sourceEpochIsSafe && targetEpochIsSafe) {
+        LOG.error(
+            "Refusing to sign attestation because source epoch "
+                + sourceEpoch
+                + " is less than the previously signed source epoch "
+                + attestationSourceEpoch);
+      } else if (sourceEpochIsSafe) {
+        LOG.error(
+            "Refusing to sign attestation because target epoch "
+                + targetEpoch
+                + " is less than or equal to the previously signed target epoch "
+                + attestationTargetEpoch);
+      } else {
+        LOG.error(
+            "Refusing to sign attestation because source epoch "
+                + sourceEpoch
+                + " is less than the previously signed source epoch "
+                + attestationSourceEpoch
+                + " and target epoch "
+                + targetEpoch
+                + " is less than or equal to the previously signed target epoch "
+                + attestationTargetEpoch);
+      }
```

Comment:
(nit) You could probably condense to one statement like:
```
LOG.error("Refusing to sign attestation:  source epoch ({}) is {}, target epoch ({}) is {}.  Previously signed source epoch {}, target epoch {}", sourceEpoch, sourceEpochIsSafe ? "safe" :  "unsafe", targetEpoch, targetEpochIsSafe ? "safe" :  "unsafe", attestationSourceEpoch, attestationTargetEpoch);
```

I'd also use string interpolation ({}) instead of concatenating.

Reply:
That's pretty neat. Thank you